packages/rulesets/src/spectral/test/avoid-additional-properties.test.ts

```diff
@@ -0,0 +1,224 @@
+              },
+            },
+          },
+        },
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(5)
+    expect(results[0].path.join(".")).toBe("definitions.This")
+    expect(results[1].path.join(".")).toBe("definitions.That.properties.nonTags")
+    expect(results[2].path.join(".")).toBe("definitions.ThaOther.properties")
+    expect(results[3].path.join(".")).toBe("definitions.Other.properties")
+    expect(results[4].path.join(".")).toBe("definitions.ThisOther.properties.tags.nonTags")
+    expect(results[0].message).toBe(errorMessage)
+    expect(results[1].message).toBe(errorMessage)
+    expect(results[2].message).toBe(errorMessage)
+    expect(results[3].message).toBe(errorMessage)
+    expect(results[4].message).toBe(errorMessage)
+  })
+})
+
+test("AvoidAdditionalProperties should find no errors", () => {
+  const oasDoc = {
+    swagger: "2.0",
+    definitions: {
+      This: {
+        description: "This",
+        type: "object",
+        properties: {
+          type: "object",
+          tags: {
+            type: "object",
+            additionalProperties: {
+              type: "object",
+              params: {
+                type: "boolean",
+              },
+            },
+          },
+        },
+      },
+      That: {
+        description: "That",
+        type: "object",
+        properties: {
+          tags: {
+            type: "object",
+            additionalProperties: {
+              type: "boolean",
+            },
+          },
+          params: {
+            type: "boolean",
+          },
+        },
+      },
+      ThaOther: {
+        description: "ThaOther",
+        type: "object",
+        properties: {
+          tags: {
+            type: "object",
+            info: {
+              type: "String",
+            },
+          },
+        },
+      },
+      Other: {
+        description: "Other object",
+        type: "object",
+        properties: {
+          type: "string",
+        },
+        tags: {
+          type: "object",
+          additionalProperties: {
+            type: "object",
+            params: {
+              type: "boolean",
+            },
+          },
+        },
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+
+test("AvoidAdditionalProperties similar to swagger should find no errors", () => {
+  const oasDoc1 = {
+    swagger: "2.0",
+    definitions: {
+      type: "object",
+      source: {
+        type: "object",
+        properties: {
+          type: "object",
+          x: {
+            type: "object",
+            $ref: "#/definitions/info",
+          },
+        },
+      },
+      info: {
+        type: "object",
+        tags: {
+          type: "object",
+          additionalProperties: {
+            type: "object",
+            params: {
+              type: "boolean",
+            },
+          },
```

Comment:
This is not a good representation of swagger. In an actual swagger file, info will have properties underneath it and then the tags would show up under properties. Can u use an example that closely represents an RP swagger?

Reply:
I added tags under properties now to represent swagger, but either case the given expression will check for "additionalProperties" if it has parent as "tags" irrespective of where it's defined.

Reply:
yes I understand the logic. But lets make sure to stick to a proper swagger def in all our tests